[vcpkg] powershell-core is required by vcpkg_copy_tool_dependencies on non-Windows, but doesn't make powershell-core available

**Host Environment**
- Archlinux (up to date as of 02/11/2020)
- Compiler: gcc (GCC) 10.2.0

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg --triplet x64-linux install cairo`

**Failure logs** 

```
20-11-02 18:07:38 abique@knot:~/d/vcpkg git:master> ./vcpkg --triplet x64-linux install cairo
Computing installation plan...
The following packages will be built and installed:
  * brotli[core]:x64-linux
  * bzip2[core]:x64-linux
    cairo[core]:x64-linux
  * dirent[core]:x64-linux
  * expat[core]:x64-linux
  * fontconfig[core]:x64-linux
  * freetype[bzip2,core,png]:x64-linux
  * libiconv[core]:x64-linux
  * libpng[core]:x64-linux
  * pixman[core]:x64-linux
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-linux...
Starting package 1/10: brotli:x64-linux
Building package brotli[core]:x64-linux...
Could not locate cached archive: /home/abique/.cache/vcpkg/archives/d4/d4e29d565a5af427af69da25079f3fe8be3f0803.zip
-- Using cached /home/abique/develop/vcpkg/downloads/google-brotli-e61745a6b7add50d380cfd7d3883dd6c62fc2c71.tar.gz
-- Cleaning sources at /home/abique/develop/vcpkg/buildtrees/brotli/src/6c62fc2c71-a8c4ea9278.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/abique/develop/vcpkg/downloads/google-brotli-e61745a6b7add50d380cfd7d3883dd6c62fc2c71.tar.gz
-- Applying patch install.patch
-- Applying patch fix-arm-uwp.patch
-- Applying patch pkgconfig.patch
-- Using source at /home/abique/develop/vcpkg/buildtrees/brotli/src/6c62fc2c71-a8c4ea9278.clean
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
CMake Error at scripts/cmake/vcpkg_copy_tool_dependencies.cmake:22 (message):
  Could not find PowerShell Core; please open an issue to report this.
Call Stack (most recent call first):
  ports/brotli/portfile.cmake:23 (vcpkg_copy_tool_dependencies)
  scripts/ports.cmake:135 (include)


Error: Building package brotli:x64-linux failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: brotli:x64-linux
  Vcpkg version: 2020.06.15-unknownhash

Additionally, attach any relevant sections from the log files above.
20-11-02 18:08:02 1 abique@knot:~/d/vcpkg git:master> 

```

**Additional context**
I don't know why it is looking for PowerShell.

--- a/scripts/cmake/vcpkg_copy_tool_dependencies.cmake
+++ b/scripts/cmake/vcpkg_copy_tool_dependencies.cmake
@@ -17,23 +17,25 @@
 ## * [glib](https://github.com/Microsoft/vcpkg/blob/master/ports/glib/portfile.cmake)
 ## * [fltk](https://github.com/Microsoft/vcpkg/blob/master/ports/fltk/portfile.cmake)
 function(vcpkg_copy_tool_dependencies TOOL_DIR)
-    find_program(PWSH_EXE pwsh)
-    if (NOT PWSH_EXE)
-        message(FATAL_ERROR "Could not find PowerShell Core; please open an issue to report this.")
+    if (VCPKG_TARGET_IS_WINDOWS)
+        find_program(PWSH_EXE pwsh)
+        if (NOT PWSH_EXE)
+            message(FATAL_ERROR "Could not find PowerShell Core; please open an issue to report this.")
+        endif()
+        macro(search_for_dependencies PATH_TO_SEARCH)
+            file(GLOB TOOLS "${TOOL_DIR}/*.exe" "${TOOL_DIR}/*.dll")
+            foreach(TOOL IN LISTS TOOLS)
+                vcpkg_execute_required_process(
+                    COMMAND "${PWSH_EXE}" -noprofile -executionpolicy Bypass -nologo
+                        -file "${SCRIPTS}/buildsystems/msbuild/applocal.ps1"
+                        -targetBinary "${TOOL}"
+                        -installedDir "${PATH_TO_SEARCH}"
+                    WORKING_DIRECTORY "${VCPKG_ROOT_DIR}"
+                    LOGNAME copy-tool-dependencies
+                )
+            endforeach()
+        endmacro()
+        search_for_dependencies("${CURRENT_PACKAGES_DIR}/bin")
+        search_for_dependencies("${CURRENT_INSTALLED_DIR}/bin")
     endif()
-    macro(search_for_dependencies PATH_TO_SEARCH)
-        file(GLOB TOOLS "${TOOL_DIR}/*.exe" "${TOOL_DIR}/*.dll")
-        foreach(TOOL IN LISTS TOOLS)
-            vcpkg_execute_required_process(
-                COMMAND "${PWSH_EXE}" -noprofile -executionpolicy Bypass -nologo
-                    -file "${SCRIPTS}/buildsystems/msbuild/applocal.ps1"
-                    -targetBinary "${TOOL}"
-                    -installedDir "${PATH_TO_SEARCH}"
-                WORKING_DIRECTORY "${VCPKG_ROOT_DIR}"
-                LOGNAME copy-tool-dependencies
-            )
-        endforeach()
-    endmacro()
-    search_for_dependencies("${CURRENT_PACKAGES_DIR}/bin")
-    search_for_dependencies("${CURRENT_INSTALLED_DIR}/bin")
 endfunction()

